hawkes/thinning_optimize.py

```diff
@@ -0,0 +1,649 @@
+                if s-timeseries[left][0]>=10:
+                    left+=1 
+                else:
+                    break
+            for point in timeseries[left:]: #point is of the form(s, k)
+                kern=powerLawCutoff(time=s-point[0], alpha=params[0][0][point[1]]*params[0][1][point[1]], beta=params[0][2][point[1]], gamma=params[0][3][point[1]])
+                kern=kern.reshape((12, 1))
+                decays+=todmult*kern
+        #print(decays.shape) #should be (12, 1)
+        decays=np.maximum(decays, 0)
+        decays[5] = ((spread/avgSpread)**beta)*decays[5]
+        decays[6] = ((spread/avgSpread)**beta)*decays[6]
+        if 100*np.round(spread, 2)  < 2 : 
+            decays[5] = decays[6] = 0
+        lamb = float(sum(decays))
+        #print("LAMBDA: ", lamb)
+        
+        """Testing candidate point"""
+        D=np.random.uniform(0, 1)
+        #print("Candidate D: ", D)
+        if D*lamb_bar<=lamb:
+            """Accepted so assign candidate point to a process by a ratio of intensities"""
+            k=0
+            total=decays[k]
+            while D*lamb_bar >= total:
+                k+=1
+                total+=decays[k]
+            """dimension is cols[k]"""   
+            """Update values of lambda for next simulation loop and append point to Ts"""
+            if k in [5, 6]:
+                spread=spread-0.01
+            
+            """Precalc next value of lambda_bar"""    
+            newdecays=todmult * params[0][0][k].reshape(12, 1)*params[0][1][k].reshape(12, 1)
+            newdecays=np.maximum(newdecays, 0)
+            newdecays[5] = ((spread/avgSpread)**beta)*newdecays[5]
+            newdecays[6] = ((spread/avgSpread)**beta)*newdecays[6]
+            if 100*np.round(spread, 2) < 2 : newdecays[5] = newdecays[6] = 0
+            lamb+= sum(newdecays)
+            lamb=lamb[0]
+
+
+            if len(Ts[k]) > 0:
+                T_Minus1 = Ts[k][-1]
+            else:
+                T_Minus1 = 0
+            decays = np.array(baselines.copy())
+            hourIndex = np.min([12,int(np.floor(s/1800))])
+            decays[5] = ((spread/avgSpread)**beta)*decays[5]
+            decays[6] = ((spread/avgSpread)**beta)*decays[6]
+            decays = decays*(s-T_Minus1)
+            
+            """Updating history and returns"""
+            Ts[k]+=(s,)
+            tau = decays[k][0]
+            numJumps+=1
+            n[k]+=1
+            timeseries.append((s, k)) #(time, event)
+            #print("point added")
+            if numJumps>=maxJumps:
+                return s, n, Ts, tau, lamb, timeseries, left
+    return s, n, Ts, -1, lamb, timeseries, left
+# time=s-tau
+# alpha=kernelParams[0]*kernelParams[1][0]
+# beta=kernelParams[1][1]
+# gamma=kernelParams[1][2]
+#%%
+def simulate2(T , paramsPath , todPath, s0 = None, filePathName = None, Pis = None, Pi_Q0 = None, beta = 0.7479, avgSpread = 0.0169, spread0 = 3, price0 = 260):
```

Comment:
rename to simulate_optimized and rename file to simulate_optimized as well. change the runner.ipynb notebook to run and import this function rather than old simulate. let's keep the old simulate.py for correctness checks if we run into some problems later.